pkgs/development/haskell-modules/hackage-packages.nix

```diff
@@ -144090,275 +144090,89 @@ self: {
-       description = "Signatures for support tensors in hasktorch";
-       license = lib.licenses.bsd3;
-       hydraPlatforms = lib.platforms.none;
-       broken = true;
-     }) {};
-
-  "hasktorch-signatures-types" = callPackage
-    ({ mkDerivation, base, deepseq }:
-     mkDerivation {
-       pname = "hasktorch-signatures-types";
-       version = "0.0.1.0";
-       sha256 = "0zaa0ihgbsiwqla46dixmxki75miy5dz91agvvd147rmr2khx1j2";
-       revision = "1";
-       editedCabalFile = "0da2sv2cahv05cymh4285s35y4b6snrab62zaibnnqbd0nk55qka";
-       libraryHaskellDepends = [ base deepseq ];
-       doHaddock = false;
-       description = "Core types for Hasktorch backpack signatures";
-       license = lib.licenses.bsd3;
-     }) {};
-
-  "hasktorch-types-th" = callPackage
-    ({ mkDerivation, base, c2hs, inline-c }:
-     mkDerivation {
-       pname = "hasktorch-types-th";
-       version = "0.0.1.0";
-       sha256 = "0irlf1lvadnr3j3zjakvkvrwdw8gpg5smk69w9l54idwsi6yvhdd";
-       revision = "1";
-       editedCabalFile = "0zgz7l8nawpjrc4p43xxfh9brl0mpszdxgahsn9977q5z08h4wnd";
-       libraryHaskellDepends = [ base inline-c ];
-       libraryToolDepends = [ c2hs ];
-       description = "C-types for Torch";
+       librarySystemDepends = [ c10 torch torch_cpu ];
+       testHaskellDepends = [ base hspec safe-exceptions ];
+       homepage = "https://github.com/hasktorch/hasktorch#readme";
+       description = "Haskell bindings for PyTorch";
        license = lib.licenses.bsd3;
-     }) {};
-
-  "hasktorch-types-thc" = callPackage
-    ({ mkDerivation, base, c2hs, hasktorch-types-th, inline-c }:
-     mkDerivation {
-       pname = "hasktorch-types-thc";
-       version = "0.0.1.0";
-       sha256 = "06jxjn9s34myy4v8ad42xqmkyad5qraj99a3vpcxfagjxwcn4hbd";
-       libraryHaskellDepends = [ base hasktorch-types-th inline-c ];
-       libraryToolDepends = [ c2hs ];
-       description = "C-types for Cutorch";
-       license = lib.licenses.bsd3;
-     }) {};
-
-  "hasktorch-zoo" = callPackage
-    ({ mkDerivation, backprop, base, deepseq, dimensions, directory
-     , filepath, generic-lens, ghc-typelits-natnormalise, hashable
-     , hasktorch, JuicyPixels, microlens, mtl, mwc-random, primitive
-     , safe-exceptions, singletons, transformers, vector
-     }:
-     mkDerivation {
-       pname = "hasktorch-zoo";
-       version = "0.0.1.0";
-       sha256 = "1cpk2q1m68y7wljaki1d4a4y45hqh34ia8r6zfw0b62f9b6zihjm";
-       libraryHaskellDepends = [
-         backprop base deepseq dimensions directory filepath generic-lens
-         ghc-typelits-natnormalise hashable hasktorch JuicyPixels microlens
-         mtl mwc-random primitive safe-exceptions singletons transformers
-         vector
-       ];
-       description = "Neural architectures in hasktorch";
+       configureFlags = [
+         "--extra-include-dirs=${lib.getDev pkgs.libtorch-bin}/include/torch/csrc/api/include"
+       ]  ++ lib.optionals pkgs.config.cudaSupport [ "-f cuda" ];
```

Comment:
The `"-f cuda"` here relies on `__structuredAttrs = false`, I think you need to split this into two strings

Reply:
It works, but I'll split it. Thx!

Reply:
It was fixed.